core/src/main/java/org/apache/druid/data/input/impl/HttpInputSource.java

```diff
@@ -69,6 +70,13 @@ public HttpInputSource(
     this.config = config;
   }
 
+  public static void throwIfInvalidProtocols(List<URI> uris)
```

Comment:
nit:

```suggestion
  private static void throwIfInvalidProtocols(List<URI> uris)
```

Reply:
It should be public since this method is now used by `HttpFirehoseFactory` as well.

Reply:
😢